### PR TITLE
[ROS-382][1/3] Address Compiler warnings in Roscala

### PR DESCRIPTION
**Pull request by AbnerZheng**
As the ticket suggests, I divided into several PRs. This is the first PR.
@tymm  PTAL

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 1]
```diff
@@ -52,9 +52,9 @@ object Location {
 
               case GlobalVariable(offset) => pureCtxt(globalEnv.slot.lift(offset))
 
-              case BitField(indirect, level, offset, spanSize, sign) => pureCtxt(None)
+              case BitField(_, _, _, _, _) => pureCtxt(None)
```
Comment (birchmd) [inline roscala/src/main/scala/coop/rchain/rosette/Location.scala]: More of a personal preference, but I think `case _: BitField` would be better. If you're not going to use any of the extracted values, why do the extraction at all?

File: roscala/src/main/scala/coop/rchain/rosette/Location.scala [hunk 2]
```diff
@@ -52,9 +52,9 @@ object Location {
...
 
-              case BitField00(offset, spanSize, sign) => pureCtxt(None)
+              case BitField00(_, _, _) => pureCtxt(None)
```
Comment (birchmd) [inline roscala/src/main/scala/coop/rchain/rosette/Location.scala]: Ditto here.

File: roscala/src/main/scala/coop/rchain/rosette/prim/fixnum.scala [hunk 3]
```diff
@@ -475,7 +480,7 @@ object fixnum {
     }
   }
 
-  private def checkFixnum(n: Int, elem: Seq[Ob]): Either[PrimError, Fixnum] =
```
Comment (birchmd) [inline roscala/src/main/scala/coop/rchain/rosette/prim/fixnum.scala]: I don't understand the reason for removing the `private` modifier. If you're trying to get rid of a warning like "private method never called" or something along those lines then just delete the method entirely -- it's obviously needed in that case.

Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/rosette/prim/fixnum.scala]: Yes, you are right. 
Maybe it is a mistake of compiler.

File: roscala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 1 above)
Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: roscala/src/main/scala/coop/rchain/rosette/Location.scala (hunk 2 above)
Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/rosette/Location.scala]: Done

File: roscala/src/main/scala/coop/rchain/rosette/prim/fixnum.scala (hunk 3 above)
Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/rosette/prim/fixnum.scala]: Done

**Pull request merged**